internal/profile/profile.go

```diff
@@ -12,13 +14,33 @@ var (
 	ErrInvalidGrepIdx = errors.New("invalid grep index")
 )
 
+type SeverityGetter interface {
+	GetSeverity() string
+}
+
+func (r *Request) GetSeverity() string {
+	return r.IssueSeverity
+}
+
+func (r *Response) GetSeverity() string {
+	return r.IssueSeverity
+}
+
+func (a *Active) GetSeverity() string {
+	if len(a.GetSteps()) > 0 {
+		return a.GetSteps()[0].IssueSeverity
+	}
+	return ""
+}
+
 // Profile represents the behavior expected from a scan profile.
 // It can be a passive or active profile (e.g. Active).
 type Profile interface {
 	GetName() string
 	GetType() Type
 	IsEnabled() bool
 	GetTags() []string
+	GetSteps() []Step
```

Comment:
I don't understand why we need this 🤔 It doesn't look like an outstanding design, cause it's only implemented by `Active`.

If you definitely need to access `.Steps`, and you have a `Profile`, then I'd suggest to try to type convert it with:

```go
active, isActive := p.(*profile.Active)
```

However, at time of filtering profiles, you already have typed profiles, so you probably don't need type conversation, but just rethink the `SeverityGetter` interface approach, which by the way doesn't work as expected (see the other comment).